astropy/coordinates/representation.py

```diff
@@ -348,11 +376,15 @@ def __repr__(self):
         prefixstr = '    '
         arrstr = _array2string(self._values, prefix=prefixstr)
 
+        diffstr = ''
+        if getattr(self, 'differentials', None):
+            diffstr = '\n (has differentials: {0})'.format(
+                ', '.join([repr(key) for key in self.differentials.keys()]))
```

Comment:
It's clear there's not consensus here, so I'm ok with merging as it stands on this issue and will make a separate small PR to make the change I suggested.